conflicting types for lnkr_pinned_rodata_start and lnkr_pinned_rodata_end

**Describe the bug**
I'm encountering a conflicting types issue during compilation. Here's the error:
```
In file included from C:/Users/ofirs/bsp/src/main.c:29:
C:/Users/ofirs/bsp/zephyr/include/zephyr/linker/utils.h: In function 'linker_is_in_rodata':
C:/Users/ofirs/bsp/zephyr/include/zephyr/linker/utils.h:27:27: error: conflicting types for 'lnkr_pinned_rodata_start'; have 'const char[]'
   27 |         extern const char lnkr_pinned_rodata_start[];
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from C:/Users/ofirs/bsp/zephyr/include/zephyr/app_memory/app_memdomain.h:9,
                 from C:/Users/ofirs/bsp/zephyr/include/zephyr/rtio/rtio.h:31,
                 from C:/Users/ofirs/bsp/zephyr/include/zephyr/drivers/sensor.h:30,
                 from C:/Users/ofirs/bsp/include/drivers/power_monitor/power_monitor.h:10,
                 from C:/Users/ofirs/bsp/include/device_handles/power_monitor.h:13,
                 from C:/Users/ofirs/bsp/include/services/health_monitor/health_monitor.h:1,
                 from C:/Users/ofirs/bsp/include/epump/epump_protocol.h:18,
                 from C:/Users/ofirs/bsp/src/main.c:13:
C:/Users/ofirs/bsp/zephyr/include/zephyr/linker/linker-defs.h:308:13: note: previous declaration of 'lnkr_pinned_rodata_start' with type 'char[]'
  308 | extern char lnkr_pinned_rodata_start[];
      |             ^~~~~~~~~~~~~~~~~~~~~~~~
C:/Users/ofirs/bsp/zephyr/include/zephyr/linker/utils.h:28:27: error: conflicting types for 'lnkr_pinned_rodata_end'; have 'const char[]'
   28 |         extern const char lnkr_pinned_rodata_end[];
      |                           ^~~~~~~~~~~~~~~~~~~~~~
C:/Users/ofirs/bsp/zephyr/include/zephyr/linker/linker-defs.h:309:13: note: previous declaration of 'lnkr_pinned_rodata_end' with type 'char[]'
  309 | extern char lnkr_pinned_rodata_end[];
      |             ^~~~~~~~~~~~~~~~~~~~~~
[118/379] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/ring_buffer.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' --build 'C:\Users\ofirs\bsp\build'
```
It seems the same symbols, `lnkr_pinned_rodata_start` and `lnkr_pinned_rodata_end`, are declared with conflicting types in:
1. zephyr/include/zephyr/linker/utils.h (const char[])
2. zephyr/include/zephyr/linker/linker-defs.h (char[])

**To Reproduce**  
1. Enable `CONFIG_LINKER_USE_PINNED_SECTION`.  
2. Include both `zephyr/include/zephyr/linker/utils.h` and `zephyr/include/zephyr/linker/linker-defs.h` in your code.  
3. Define and use a pinned section in your custom linker script.  
4. Attempt to build the project.  

**Expected Behavior**  
The code should compile without conflicting type errors between the two headers.

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain Zephyr SDK 0.16.8
 - Commit SHA: 305ceba9204d9a7ee4bae0a6103ef4cbd9a1c54e

--- a/include/zephyr/linker/utils.h
+++ b/include/zephyr/linker/utils.h
@@ -24,8 +24,8 @@
 static inline bool linker_is_in_rodata(const void *addr)
 {
 #if defined(CONFIG_LINKER_USE_PINNED_SECTION)
-	extern const char lnkr_pinned_rodata_start[];
-	extern const char lnkr_pinned_rodata_end[];
+	extern char lnkr_pinned_rodata_start[];
+	extern char lnkr_pinned_rodata_end[];
 
 	if (((const char *)addr >= (const char *)lnkr_pinned_rodata_start) &&
 	    ((const char *)addr < (const char *)lnkr_pinned_rodata_end)) {

